src/parser.ts

```diff
@@ -353,6 +353,7 @@ function parseSchema(
     ast: parse(value, options, key, processed, usedNames),
     isPatternProperty: false,
     isRequired: includes(schema.required || [], key),
+    isReadOnly: !!value.readOnly,
```

Comment:
Let's try to be stricter here:

1. Add a normalizer step to add `readOnly: false` to each schema if it's not already defined
2. Update the type of `NormalizedJSONSchema` to reflect that `readOnly` is guaranteed to be defined
3. Update your changes to parser.ts to directly read `readOnly`, and avoid the double cast (`!!`)

Reply:
done in 6d4eec5

Reply:
I've started to go this way but I'm struggling a bit to define [when to add this property to the schema or not](https://github.com/bcaudan/json-schema-to-typescript/commit/253a975883a4ebc2727298215d003d44bd0633ec#diff-f8b5c2873187ca697b26451706f58782775ac1c2656fc73e4fc1d8193febfe36R137).
Am I missing something?

Reply:
You can use the `isObjectSchema` and `isArraySchema` utils to help (see other normalizer steps for examples).